File: C/1040.c
#include <stdio.h>

int main() {
    double nota1, nota2, nota3, nota4;

    scanf("%lf %lf %lf %lf", &nota1, &nota2, &nota3, &nota4);

    double media = ((nota1 * 2) + (nota2 * 3) + (nota3 * 4) + (nota4 * 1)) / 10;

    printf("Media: %.1lf\n", media);

    if( media >= 7.0 )
        printf("Aluno aprovado.\n");

    else if( media < 5.0 )
        printf("Aluno reprovado.\n");

    else {
        printf("Aluno em exame.\n");
        double exame;
        scanf("%lf", &exame);
        printf("Nota do exame: %.1lf\n", exame);

        double newMedia = (media + exame) / 2;

        if( newMedia >= 5.0 )
            printf("Aluno aprovado.\n");

        else
            printf("Aluno reprovado.\n");

        printf("Media final: %.1lf\n", newMedia);
    }

    return 0;
}

File: JavaScript/1040.js
let input = require('fs').readFileSync('./dev/stdin/file.txt', 'utf8');
let lines = input.split('\n');
let notas = lines[0].split(' ');

let n1 = parseFloat(notas.shift());
let n2 = parseFloat(notas.shift());
let n3 = parseFloat(notas.shift());
let n4 = parseFloat(notas.shift());

let media = (n1 * 2 + n2 * 3 + n3 * 4 + n4) / 10;
console.log(`Media: ${media.toFixed(1)}`);

if( media >= 7 )
    console.log('Aluno aprovado.');

else if( media >= 5 && media <= 6.9 ) {
    console.log('Aluno em exame.')
    
    let notaExame = parseFloat(lines[1]);
    let mediaFinal = (notaExame + media) / 2;

    console.log(`Nota do exame: ${notaExame.toFixed(1)}`);
    
    if( mediaFinal >= 5 )
        console.log('Aluno aprovado.');

    else
        console.log('Aluno reprovado.');

    console.log(`Media final: ${mediaFinal.toFixed(1)}`);
}

else
    console.log('Aluno reprovado.');